### PR TITLE
Update: Group samples view

### PR DESCRIPTION
**Pull request by deepsidhu85**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR updated group samples view to display samples that the user has access to through namespace group links. This was accomplished by adding a sample policy which returns samples for projects within a group as well as samples from projects within namespace group links.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

image

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Login as admin@email.com
2. Visit the group page for `Bacillus` (or any other group which has namespaces linked to it)
3. Go through the samples and you should see samples from projects that you have access to through namespace group links
4. Create 3 groups, `Group A`, `Group B`, and `Group C`
5. Create a project under  the groups from step 4
6. Create a few samples per project from step 5
7. Go to the Members page of the project created under Group A  and click on the `Groups` tab
8. Invite `Group B` with any group access level
9. Go to the Members page for Group A, and click on the `Groups` tab
10. Invite `Group C`
11. Go to the samples page of `Group B`. You should see samples from from the project in Group A plus any samples from projects under `Group B`
12. Go to the samples page of `Group C`. You should see samples from any projects under `Group A` plus any samples from projects under `Group C`
13. Ensure tests pass locally

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Review by ChrisHuynh333 — APPROVED** [review]
Nothing to add code-wise. It's working for me as described.

**Review by ChrisHuynh333 — APPROVED** [review]
Working as described, nothing to add.

File: app/policies/sample_policy.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+# Policy samples authorization
+class SamplePolicy < ApplicationPolicy
+  scope_for :relation, :group_samples do |relation, options|
+    group = options[:group]
```
Comment (ericenns) [inline app/policies/sample_policy.rb:6]: **nitpick (blocking):** Can you add the following after this line, to only return `Samples` when the `User` has permission, just as another failsafe.

```ruby
next scope.none if !Member.can_view?(user, group)
```

**Review by ericenns — CHANGES_REQUESTED** [review]
Looks good, one minor comment before acceptance

File: app/policies/sample_policy.rb (hunk 1 above)
Comment (deepsidhu85) [reply app/policies/sample_policy.rb:6]: Good call. Updated in 2c93588ea584bafca3f264f876280279799314d1

**Review by ericenns — APPROVED** [review]
Looks good to me!

**Pull request merged**